content/blog/2017/2017-02-02-scm-api-2-take2.adoc

```diff
@@ -0,0 +1,124 @@
+---
+:layout: post
+:title: SCM API 2.0 Release Take 2
+:tags:
+- development
+- plugins
+:author: stephenc
+---
+
+In January we link:https://jenkins.io/blog/2017/01/17/scm-api-2/[announced the release of SCM API 2.0].
+There were four important issues identified after the original release was published and in order to allow those issues to be resolved, the updates were pulled from the update center.
+
+The issues have now been resolved and the plugins are now available from the update center.
+
+== Summary for busy Jenkins Administrators
+
+TIP: Always take a backup of your `JENKINS_HOME` before upgrading any plugins.
+
+[WARNING]
+====
+If you upgrade to Branch API 2.0.x and you have either the GitHub Branch Source or the BitBucket Branch Source plugins *and* you do not upgrade those instances to the 2.0.x line then your Jenkins instance will fail to start-up correctly.
+
+The solution is just to upgrade the GitHub Branch Source or the BitBucket Branch Source plugin (as appropriate) to the 2.0.x line.
+====
+
+Upgrading should make multi-branch projects much better.
+When you are ready to upgrade you must ensure that you upgrade all the required plugins.
+If you miss some, just upgrade them and restart to fix the issue.
+
+NOTE: _Version numbers in *bold* indicate a change from the original version in the link:https://jenkins.io/blog/2017/01/17/scm-api-2/[original announcment]_
+
+Folders Plugin:: *5.17* or newer
+SCM API Plugin:: *2.0.2* or newer
+Branch API Plugin:: *2.0.2* or newer
+Git Plugin:: This depends on the exact release line of the Git plugin that you are using.
++
+* Following the 2.6.x release line: *2.6.4* or newer
+* Following the 3.0.x release line (_recommended_): *3.0.4* or newer
+
+Mercurial Plugin:: 2.0.0 or newer
+GitHub Branch Source Plugin:: *2.0.1* or newer
+BitBucket Branch Source Plugin:: *2.0.2* or newer
+GitHub Organization Folders Plugin:: 1.6
+Pipeline Multibranch Plugin:: *2.12* or newer
+
+If you are using the link:https://plugins.jenkins.io/blueocean[Blue Ocean plugin]
+
+Blue Ocean Plugin:: *1.0.0-b22* or newer
+
+Other plugins that may require updating:
+
+GitHub API Plugin:: 1.84 or newer
+GitHub Plugin:: 1.25.0 or newer
+
+After an upgrade you will see the data migration warning (see the screenshot in link:https://issues.jenkins-ci.org/browse/JENKINS-41608[JENKINS-41608] for an example) this is normal and expected.
+The unreadable data will be removed by the next scan / index or can be removed manually using the _Discard Unreadable Data_ button.
+The warning will disappear on the next restart after the unreadable data has been removed.
+
+== Analysis of the issues
+
+link:https://issues.jenkins-ci.org/browse/JENKINS-41121[JENKINS-41121]::
```

Comment:
Maybe have the title/summary of the JIRA here as well? 

`JENKINS-41121: GitHub Branch Source upgrade can cause a lot of rebuilds`